lib/matplotlib/collections.py

```diff
@@ -2016,6 +2016,29 @@ def set_paths(self):
             self._meshWidth, self._meshHeight, self._coordinates)
         self.stale = True
 
+    def set_array(self, A):
+        if isinstance(self, QuadMesh) and A is not None:
+            A_ = A.ravel()
+            width, height = self._meshWidth, self._meshHeight
+            if self._shading == 'flat':
+                if not (len(A_) == width * height or
```

Comment:
Um, don't you have to check width and height separately? Otherwise, the dfiffernce between (3, 4) and (4, 3) would not be catched.

Reply:
Ah, yikes I forgot about that case. My apologies, I'll fix this.

Reply:
The simplest way seemed to be doing it with a flag, otherwise we'd have to repeat type error multiple times